### ISSUE TITLE
Package league/url is abandoned, you should avoid using it. Use league/uri instead.

### ISSUE THREAD

**Issue by skornev** [Payum/Payum#503]
Can we replace it with  league/uri instead?

**Comment by makasim** [comment]
same as https://github.com/Payum/Payum/issues/482

**Comment by skornev** [comment]
But can we use it 2? Because BC is already broken in it as far as I am aware.

**Comment by makasim** [comment]
> Because BC is already broken in it as far as I am aware.

Where?

**Comment by makasim** [comment]
Okay, I think we can switch to a new version. The url classes is used internally in payum so it should be safe to upgrade and will not affect anybody. 

@skornev will you work on PR?

**Comment by skornev** [comment]
AddPaymentFactory was replaced with addGatewayFactory and then removed. So this code for some bundles won't work anymore.

```
namespace AcmeBundle;

use Symfony\Component\DependencyInjection\ContainerBuilder;
use Symfony\Component\HttpKernel\Bundle\Bundle;
use Payum\Bundle\PayumBundle\DependencyInjection\PayumExtension;
use Skornev\Payum\Acme\AcmeGatewayFactory;




class AcmeBundle extends Bundle
{
    public function build(ContainerBuilder $container)
    {
        parent::build($container);
        /** @var  PayumExtension $extension */
        $extension = $container->getExtension('payum');
        $extension->addGatewayFactory(new AcmeGatewayFactory());


    }
}
```

> 
Attempted to call an undefined method named "addGatewayFactory" of class "Payum\Bundle\PayumBundle\DependencyInjection\PayumExtension"

Everything works find with gateway_factory_builder and tags.

**Comment by skornev** [comment]
and yes for the PR. I ll try.

**Comment by makasim** [comment]
> Everything works fine with gateway_factory_builder and tags.

The `addGatewayFactory ` is for the bundle version 1.x. The builder is for version 2.x. I suspect you are using the 2.x

**Comment by skornev** [comment]
Yes. That's true. I use 2x. I know that's issue is more related to the bundle and it was marked as deprecated. Even it's still used to use  tags inside, for me it's a kind of BC, since this method was removed.
Anyway,  it's more related to the bundle rather than payum itself. 
As for payum BC.
For example FillOrderDetails was marked as deprecated and replaced with Convert. Later it was removed completely. 
I mean BC has been broken many times, so I don't see any reason not to do that in 2.x during transition from league/url to league/uri if it's needed or  maybe it won't and transition  go smoothly.

**Comment by makasim** [comment]
fix merged to master. will be available since 1.3.0
